backend/cms/models/lehrstellenradar_extra.py

```diff
@@ -0,0 +1,25 @@
+from cms.models.extra import Extra
+
+
+class LehrstellenradarExtra(Extra):
```

Comment:
Yes, it may be a little bit over-engineered what I did here, but I still think it's appropriate :D
My model solves the problem, that we won't need to introduce strange attributes on the Site-model for determining what the extra should look like in the api (for example the wohnraumboerse-name, wohnraumboerse-url, wohnraumboerse-api, wohnraumboerse-thumbnail attributes in wp-cms just for one extra that is activated for two Sites), but have them defined in a relation table of wohnraumboerse-extra and Site. That way we don't pollute the Site model unnecessarily.

Also making them all just one unified extra would mean, that we have to hardcode their behaviour somewhere else (probably in the api then)...

What do the others think about that?

Reply:
Fully agree

Reply:
Ok, the wohnraumboerse is the only extra which would justify the division into all the extra subclasses. But I just came up with another idea how to solve this problem: What if we saved the two wohnraumboerse extras as two different extras (one called wohnraumboerse, the other one raumfrei) in the database and just activate one of them on the desired sites? This way we won't neither need the additional attributes in the site-model, nor the complex model-structure with hardcoded values. Additionally, it keeps the maintenance effort low because we can add/change/delete extras in the backend without changing the code.

Reply:
Yeah, I share your concern that we don't want to have 10+ different extras for only a single site. So even if it is unlikely that we're going to have a widespread and configurable extra in the future, we need some kind of plan B what we do if this happens, right? What about an `ExtraSetting`-model which is associated to exactly one extra and one site? If I'm not mistaken, we could theoretically add this in the future without breaking changes to the existing extras, right?

Reply:
Hmm. I thought there would be a fancy way in django to merge the extras with the calling site's postal code when accessing it, but apparently there is not. So I think you were right in the first place and the cleaner model structure comes at the cost of some weird string replacements in the api or something similar. Sorry for the confusion! :sweat_smile: 
However, I'm still not 100% happy with the hardcoded urls, so I tried a more generic approach in the `feature/design-extra-models`-branch. Check it out if you're interested.